### PR TITLE
BAEL-3336: Programmatic Transaction Management

### PR DESCRIPTION
**Pull request by alimate**
<del>I chose this module since it already configured the PostgreSQL test container. Also, I had to convert my code to JUnit 4. I guess we can migrate to JUnit 5 later.</del>

### CODE REVIEW

File: persistence-modules/spring-data-jpa-3/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+=========
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa-3/README.md]: You can omit this file for now. I will add it after the article goes live.

File: persistence-modules/spring-data-jpa-3/pom.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        <dependency>
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-launcher</artifactId>
+            <version>${junit-platform.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <properties>
+        <testcontainers.version>1.10.6</testcontainers.version>
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa-3/pom.xml]: The latest is 1.12.2

File: persistence-modules/spring-data-jpa-3/pom.xml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        <postgresql.version>42.2.5</postgresql.version>
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa-3/pom.xml]: latest is 42.2.8

File: persistence-modules/spring-data-jpa-3/README.md (hunk 1 above)
Comment (alimate) [reply persistence-modules/spring-data-jpa-3/README.md]: Resolved.

File: persistence-modules/spring-data-jpa-3/pom.xml (hunk 3 above)
Comment (alimate) [reply persistence-modules/spring-data-jpa-3/pom.xml]: Resolved.

File: persistence-modules/spring-data-jpa-3/pom.xml (hunk 2 above)
Comment (alimate) [reply persistence-modules/spring-data-jpa-3/pom.xml]: Resolved.

**Pull request merged**